### PR TITLE
added testnet20 changes

### PR DESCRIPTION
**Pull request by BatiGencho**

### CODE REVIEW

File: examples/smartContracts/index.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ const ora = require('ora');
 const MASSA_EXEC_ERROR = 'massa_execution_error';
 
 const DEPLOYER_SECRET_KEY =
-  'S1NA786im4CFL5cHSmsGkGZFEPxqvgaRP8HXyThQSsVnWj4tR7d';
+  'S12srNEAvZrTb9pktYeuePpuM4taW5dfmWAZYtdqyETWTBspkoT1';
```
Comment (Thykof) [inline examples/smartContracts/index.ts:33]: it's a bad practice to have secrets in the code

File: src/interfaces/IOperationData.ts [hunk 2]
```diff
@@ -13,5 +14,9 @@ export interface IOperationData {
       sender_public_key: string;
     };
     signature: string;
+    content_creator_pub_key: string;
+    content_creator_address: string;
+    id: string; //
```
Comment (Thykof) [inline src/interfaces/IOperationData.ts:19]: either fill the comment or remove `//`

```suggestion
    id: string;
```

**Review by Thykof — CHANGES_REQUESTED** [review]

File: src/interfaces/IOperationData.ts [hunk 3]
```diff
@@ -4,7 +4,8 @@ export interface IOperationData {
   id: string; // Operation id
   in_blocks: string[]; // Block ids
   in_pool: boolean; //
```
[the remark this answers is not in the document]
Comment (BatiGencho) [reply src/interfaces/IOperationData.ts]: annotated

File: src/interfaces/IOperationData.ts (hunk 2 above)
Comment (BatiGencho) [reply src/interfaces/IOperationData.ts]: annotated

File: src/web3/BaseClient.ts [hunk 4]
```diff
@@ -263,9 +263,13 @@ export class BaseClient {
...
         // target address
-        const targetAddressEncoded = base58Decode(
-          (data as ICallData).targetAddress.slice(1),
+        let targetAddressEncoded = base58Decode(
+          (data as ICallData).targetAddress.slice(2),
         ).slice(1);
+        targetAddressEncoded = Buffer.concat([
+          Buffer.from([1]),
+          targetAddressEncoded,
+        ]);
```
[the remark this answers is not in the document]
Comment (BatiGencho) [reply src/web3/BaseClient.ts]: refactored and pulled out into a smaller func

File: examples/wallet/index.ts [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+import { EventPoller } from '../../src/web3/EventPoller';
+import { ITransactionData } from '../../src/interfaces/ITransactionData';
+import { EOperationStatus } from '../../src/interfaces/EOperationStatus';
+import { IRollsData } from '../../src/interfaces/IRollsData';
+const path = require('path');
+const chalk = require('chalk');
+const ora = require('ora');
+
+const DEPLOYER_SECRET_KEY =
+  'S12srNEAvZrTb9pktYeuePpuM4taW5dfmWAZYtdqyETWTBspkoT1';
```
[the remark this answers is not in the document]
Comment (BatiGencho) [reply examples/wallet/index.ts]: agree, I have moved the priv keys etc. to an .env file

**Review by Thykof — CHANGES_REQUESTED** [review]
1. Can you add .env file in gitignore?
2. there are duplicated code in examples, for retriving env vars, maybe it's for the example to be clearer
3. you are updating yarn lock and package lock, you have to choose :) which one yarn or npm you want to use, or maybe I miss something. For all others projects we use npm and it works well.

**Review by Thykof — APPROVED** [review]
make sure to never used the wallet of the private key in the .env.example

**Pull request merged**